src/lib/dnssd/Types.h

```diff
@@ -293,27 +297,7 @@ struct ResolvedNodeData
     }
 };
 
-struct DiscoveredNodeData
-{
-    CommonResolutionData resolutionData;
-    DnssdNodeData nodeData;
-    DiscoveryType nodeType;
-
-    void Reset()
-    {
-        resolutionData.Reset();
-        nodeData.Reset();
-        nodeType = DiscoveryType::kUnknown;
-    }
-    DiscoveredNodeData() { Reset(); }
-
-    void LogDetail() const
-    {
-        ChipLogDetail(Discovery, "Discovered node:");
-        resolutionData.LogDetail();
-        nodeData.LogDetail();
-    }
-};
+using DiscoveredNodeData = Variant<CommissionNodeData>;
```

Comment:
A variant with a single possible variant looks odd. Why do we do this? Would std::optional<> be more appropriate if we just need some way to say "has or does not have value" ?

Reply:
This PR is just for refactoring DiscoveredNodeData to Variant<CommissionNodeData>, a following PR will add OperationalNodeData to the variant.